modules/core/src/main/java/org/apache/synapse/mediators/builtin/CallMediator.java

```diff
@@ -68,55 +75,90 @@ public boolean mediate(MessageContext synInCtx) {
 
-        synLog.traceOrDebug("Start : Call mediator");
-        if (synLog.isTraceTraceEnabled()) {
-            synLog.traceTrace("Message : " + synInCtx.getEnvelope());
-        }
-
-        boolean outOnlyMessage = "true".equals(synInCtx.getProperty(
-                SynapseConstants.OUT_ONLY));
+        if (!blocking) {
+            synLog.traceOrDebug("Start : Call mediator - Non Blocking Call");
+            if (synLog.isTraceTraceEnabled()) {
+                synLog.traceTrace("Message : " + synInCtx.getEnvelope());
+            }
 
-        // Prepare the outgoing message context
-        MessageContext synOutCtx = null;
-        if (outOnlyMessage) {
-            try {
-                synOutCtx = MessageHelper.cloneMessageContext(synInCtx);
-            } catch (AxisFault axisFault) {
-                handleException("Error occurred while cloning msg context", axisFault, synInCtx);
+            //If call mediator is a marked as non blocking function
+            boolean outOnlyMessage = "true".equals(synInCtx.getProperty(SynapseConstants.OUT_ONLY));
+
+            // Prepare the outgoing message context
+            MessageContext synOutCtx = null;
+            if (outOnlyMessage) {
+                try {
+                    synOutCtx = MessageHelper.cloneMessageContext(synInCtx);
+                } catch (AxisFault axisFault) {
+                    handleException("Error occurred while cloning msg context", axisFault, synInCtx);
+                }
+            } else {
+                synOutCtx = synInCtx;
             }
-        } else {
-            synOutCtx = synInCtx;
-        }
 
-        synOutCtx.setProperty(SynapseConstants.CONTINUATION_CALL, true);
-        ContinuationStackManager.updateSeqContinuationState(synOutCtx, getMediatorPosition());
-
-        // if no endpoints are defined, send where implicitly stated
-        if (endpoint == null) {
-
-            if (synLog.isTraceOrDebugEnabled()) {
-                StringBuffer sb = new StringBuffer();
-                sb.append("Calling ").append(synOutCtx.isResponse() ? "response" : "request")
-                        .append(" message using implicit message properties..");
-                sb.append("\nCalling To: ").append(synOutCtx.getTo() != null ?
-                                                   synOutCtx.getTo().getAddress() : "null");
-                sb.append("\nSOAPAction: ").append(synOutCtx.getWSAAction() != null ?
-                                                   synOutCtx.getWSAAction() : "null");
-                synLog.traceOrDebug(sb.toString());
+            synOutCtx.setProperty(SynapseConstants.CONTINUATION_CALL, true);
+            ContinuationStackManager.updateSeqContinuationState(synOutCtx, getMediatorPosition());
+
+            // If no endpoints are defined, send where implicitly stated
+            if (endpoint == null) {
+
+                if (synLog.isTraceOrDebugEnabled()) {
+                    StringBuffer sb = new StringBuffer();
```

Comment:
I used StringBuilder instead of StringBuffer in all PRs which I sent recently. I will fix this also.
https://github.com/katheesR/esb-connectors/blob/e168cfb70b667442f65f5dc6dda2d877950605ae/twitter/twitter-connector/twitter-connector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/twitter/TwitterSignatureGeneration.java